src/customer/strategies/amazon-maxo/amazon-maxo-customer-strategy.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { getAmazonMaxo, getPaymentMethodsState } from '../../../payment/payment-methods.mock';
+import { createAmazonMaxoPaymentProcessor, AmazonMaxoPaymentProcessor } from '../../../payment/strategies/amazon-maxo';
+import { getPaymentMethodMockUndefinedMerchant } from '../../../payment/strategies/amazon-maxo/amazon-maxo.mock';
+import { RemoteCheckoutActionCreator, RemoteCheckoutRequestSender } from '../../../remote-checkout';
+import { CustomerInitializeOptions } from '../../customer-request-options';
+import { getCustomerState } from '../../customers.mock';
+import CustomerStrategy from '../customer-strategy';
+
+import AmazonMaxoCustomerStrategy from './amazon-maxo-customer-strategy';
+import { getAmazonMaxoCustomerInitializeOptions, Mode } from './amazon-maxo-customer.mock';
+
+describe('AmazonMaxoCustomerStrategy', () => {
+    let container: HTMLDivElement;
+    let customerInitializeOptions: CustomerInitializeOptions;
+    let paymentMethod: PaymentMethod;
+    let paymentProcessor: AmazonMaxoPaymentProcessor;
+    let remoteCheckoutActionCreator: RemoteCheckoutActionCreator;
+    let requestSender: RequestSender;
+    let store: CheckoutStore;
+    let strategy: CustomerStrategy;
+    let walletButton: HTMLAnchorElement;
+
+    beforeEach(() => {
+        paymentMethod = getAmazonMaxo();
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        requestSender = createRequestSender();
+
+        remoteCheckoutActionCreator = new RemoteCheckoutActionCreator(
+            new RemoteCheckoutRequestSender(requestSender)
+        );
+
+        paymentProcessor = createAmazonMaxoPaymentProcessor(store);
+
+        strategy = new AmazonMaxoCustomerStrategy(
+            store,
+            remoteCheckoutActionCreator,
+            paymentProcessor
+        );
+
+        jest.spyOn(store, 'dispatch')
+            .mockReturnValue(Promise.resolve(store.getState()));
+
+        jest.spyOn(paymentProcessor, 'initialize')
+            .mockReturnValue(Promise.resolve());
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        container = document.createElement('div');
+        container.setAttribute('id', 'amazonmaxoCheckoutButton');
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        container.appendChild(walletButton);
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    describe('#initialize()', () => {
+
+            it('Creates the button', async () => {
```

Comment:
you have 1 extra level of indentation here

Reply:
Extra level of indentation was removed.